addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -68,15 +70,16 @@ def _l10n_eg_eta_connect_to_server(self, request_data, request_url, method, is_a
                 'error': str(ex),
                 'blocking_level': 'warning'
             }
-        if not request_response.ok:
+        try:
+            if isinstance(request_response, str):
+                raise ValidationError(request_response)
```

Comment:
```suggestion
        try:
            response_data = request_response.json()
        except JsonDecodeError:
            return {'error': ...}
        
        if response_data.get('error'): 
        
        
```
What am I missing that we need more complicated?

Reply:
Hello @jco-odoo ,
- if we are not checking for this condition `if isinstance(request_response, str):` then we should handle only for `JsonDecodeError`.
- Also, for condition `if response_data.get('error'):` we must return the error with blocking level as error.
- rest we think should be fine with the flow and should work effectively.